### PR TITLE
Cross-compile for 2.13

### PR DESCRIPTION
**Pull request by JannikArndt**

### CODE REVIEW

**Comment by JannikArndt** [comment]
`scapegoat` currently fails on Scala 2.13.X. Until that is fixed in the plugin, I moved it to a seperate CircleCi step, which is _not_ activated in _this_ PR, but get's reintroduced in https://github.com/moia-dev/teleproto/pull/9

File: src/main/scala/io/moia/protos/teleproto/ReaderImpl.scala [hunk 1]
```diff
@@ -417,7 +417,7 @@ object ReaderImpl {
     val protobufOptions = symbolsByTolerantName(c)(protobufType.typeSymbol.asClass.knownDirectSubclasses.filter(_.isModuleClass))
     val modelOptions    = symbolsByTolerantName(c)(modelType.typeSymbol.asClass.knownDirectSubclasses.filter(_.isModuleClass))
 
-    val unmatchedProtobufOptions = protobufOptions.filterKeys(name => !modelOptions.contains(name)).values.map(_.name.decodedName)
+    val unmatchedProtobufOptions = protobufOptions.filterKeys(name => !modelOptions.contains(name)).toMap.values.map(_.name.decodedName)
```
Comment (markus1189) [inline src/main/scala/io/moia/protos/teleproto/ReaderImpl.scala]:
```suggestion
    val unmatchedProtobufOptions =
      protobufOptions.toList.collect { case (name, v) if !modelOptions.contains(name) => v.name.decodedName }
```

Comment (markus1189) [reply src/main/scala/io/moia/protos/teleproto/ReaderImpl.scala]: We could do it like this :point_up: and maybe avoid the deprecation warnings and also turn on the fatal warnings again

File: src/main/scala/io/moia/protos/teleproto/ReaderImpl.scala [hunk 2]
```diff
@@ -426,7 +426,7 @@ object ReaderImpl {
       )
     }
 
-    val surplusModelOptions = modelOptions.filterKeys(name => !protobufOptions.contains(name)).values.map(_.name.decodedName)
+    val surplusModelOptions = modelOptions.filterKeys(name => !protobufOptions.contains(name)).toMap.values.map(_.name.decodedName)
```
Comment (markus1189) [inline src/main/scala/io/moia/protos/teleproto/ReaderImpl.scala]:
```suggestion
    val surplusModelOptions = modelOptions
      .filter {
        case (name, _) =>
          !protobufOptions.contains(name)
      }
      .toMap
      .values
      .map(_.name.decodedName)
```

Comment (markus1189) [reply src/main/scala/io/moia/protos/teleproto/ReaderImpl.scala]: Or we could use `filter { case (key, _) => ???` instead which also works

**Comment by markus1189** [comment]
Also add the `.coursier` cache please :wink: https://github.com/moia-dev/trip-booking2/blob/ed9dde4521a908eb23ca5dcc12aa090e5658ffe0/.circleci/config.yml#L131

File: build.sbt [hunk 3]
```diff
@@ -11,13 +12,21 @@ lazy val `teleproto` =
...
+        library.scalaPBJson      % Compile,
+        library.scalaTest        % Test,
+        library.scalaCheck       % Test,
+        "org.scala-lang.modules" %% "scala-collection-compat" % "2.1.2"
+      ),
+      libraryDependencies ++= (CrossVersion.partialVersion(scalaVersion.value) match {
+        case Some((2, 13)) =>
+          Seq("org.scala-lang" % "scala-reflect" % "2.13.1")
+        case Some((2, 12)) =>
+          Seq("org.scala-lang" % "scala-reflect" % "2.12.9")
```
[the remark this answers is not in the document]
Comment (JannikArndt) [reply build.sbt]: Fixed like this: baece7b

File: src/main/scala/io/moia/protos/teleproto/ReaderImpl.scala (hunk 1 above)
Comment (JannikArndt) [reply src/main/scala/io/moia/protos/teleproto/ReaderImpl.scala]: Nice, done in c51dc1f

File: src/main/scala/io/moia/protos/teleproto/ReaderImpl.scala (hunk 2 above)
Comment (JannikArndt) [reply src/main/scala/io/moia/protos/teleproto/ReaderImpl.scala]: Nice, done in c51dc1f

**Pull request merged**